internal/proxy/multi_rate_limiter.go

```diff
@@ -138,34 +166,65 @@ func (rl *rateLimiter) limit(rt internalpb.RateType, n int) (bool, float64) {
 	return !limit.AllowN(time.Now(), n), float64(limit.Limit())
 }
 
-// setRates sets new rates for the limiters.
-func (rl *rateLimiter) setRates(rates []*internalpb.Rate) error {
-	for _, r := range rates {
+func (rl *rateLimiter) setRates(collectionRate *proxypb.CollectionRate) error {
+	log := log.With(
+		zap.Int64("proxyNodeID", paramtable.GetNodeID()),
+		zap.Int64("CollectionID", collectionRate.Collection),
+	)
+	for _, r := range collectionRate.GetRates() {
 		if limit, ok := rl.limiters.Get(r.GetRt()); ok {
 			limit.SetLimit(ratelimitutil.Limit(r.GetR()))
-			setRateGaugeByRateType(r.GetRt(), paramtable.GetNodeID(), r.GetR())
+			setRateGaugeByRateType(r.GetRt(), paramtable.GetNodeID(), collectionRate.Collection, r.GetR())
 		} else {
 			return fmt.Errorf("unregister rateLimiter for rateType %s", r.GetRt().String())
 		}
+		log.RatedInfo(30, "current collection rates in proxy",
+			zap.String("rateType", r.Rt.String()),
+			zap.Float64("rateLimit", r.GetR()),
+		)
+	}
+
+	for i := 0; i < len(collectionRate.GetStates()); i++ {
+		rl.quotaStates.Insert(collectionRate.States[i], collectionRate.Codes[i])
```

Comment:
When are states removed from quotaStates? Perhaps we should create a new quotaStates map before inserting.